File: deletephoto.php
<?php
$data = file_get_contents('php://input');
$request = json_decode($data);

if ($request && isset($request->index)) {
  $index = $request->index;
  $galleryData = json_decode(file_get_contents('gallery.json'));

  if ($index >= 0 && $index < count($galleryData->photos)) {
    array_splice($galleryData->photos, $index, 1);
    file_put_contents('gallery.json', json_encode($galleryData, JSON_PRETTY_PRINT));
  }
}
// $data = file_get_contents('php://input');
// $request = json_decode($data);

// if ($request && isset($request->photos)) {
//   $galleryData = json_decode(file_get_contents('gallery.json'));

//   foreach ($request->photos as $photo) {
//     $index = $photo->index;
//     if ($index >= 0 && $index < count($galleryData->photos)) {
//       array_splice($galleryData->photos, $index, 1);
//     }
//   }

//   file_put_contents('gallery.json', json_encode($galleryData, JSON_PRETTY_PRINT));
//   http_response_code(200); // Отправляем успешный статус ответа
// } else {
//   http_response_code(400); // Отправляем ошибку, если запрос или данные некорректны
// }
?>

File: deletephoto.py
import json
from flask import Flask, request

app = Flask(__name__)

@app.route('/deletephoto', methods=['POST'])
def delete_photo():
    data = request.get_json()
    request_data = json.loads(data)

    if request_data and 'index' in request_data:
        index = request_data['index']
        with open('gallery.json', 'r') as file:
            gallery_data = json.load(file)

        if 0 <= index < len(gallery_data['photos']):
            del gallery_data['photos'][index]

            with open('gallery.json', 'w') as file:
                json.dump(gallery_data, file, indent=4)

        return '', 200

if __name__ == '__main__':
    app.run()
